### PR TITLE
Align code style with PHPMD

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring
Breaking change: no

See https://github.com/phpmd/phpmd/pull/1138 for details

### CODE REVIEW

File: src/main/php/PDepend/Metrics/Analyzer/CodeRankAnalyzer.php [hunk 1]
```diff
@@ -113,7 +113,7 @@ class CodeRankAnalyzer extends AbstractAnalyzer implements AnalyzerNodeAware
      *
      * @var array<string, array<string, int>>
      */
-    private $nodeMetrics = null;
+    private $nodeMetrics;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/CodeRankAnalyzer.php:116]: Better to keep the default value and switch to typed property.
```suggestion
    private ?array $nodeMetrics = null;
```

File: src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php [hunk 2]
```diff
@@ -81,20 +82,20 @@ class CrapIndexAnalyzer extends AbstractAnalyzer implements AggregateAnalyzer, A
...
      * file.
      *
      * @var Report
      */
-    private $report = null;
+    private $report;
 
     /**
      * @var CyclomaticComplexityAnalyzer
      */
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php:97]:
```suggestion
```

File: src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php [hunk 3]
```diff
@@ -81,20 +82,20 @@ class CrapIndexAnalyzer extends AbstractAnalyzer implements AggregateAnalyzer, A
...
-    private $ccnAnalyzer = null;
+    private $ccnAnalyzer;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php:98]:
```suggestion
    private ?CyclomaticComplexityAnalyzer $ccnAnalyzer = null;
```

File: src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php [hunk 4]
```diff
@@ -81,20 +82,20 @@ class CrapIndexAnalyzer extends AbstractAnalyzer implements AggregateAnalyzer, A
...
+    private $metrics;
 
     /**
      * The coverage report instance representing the supplied coverage report
      * file.
      *
      * @var Report
      */
-    private $report = null;
+    private $report;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php:93]:
```suggestion
    private ?Report $report = null;
```

File: src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php [hunk 5]
```diff
@@ -81,20 +82,20 @@ class CrapIndexAnalyzer extends AbstractAnalyzer implements AggregateAnalyzer, A
...
      * @var array<string, array<string, float>>
      */
-    private $metrics = null;
+    private $metrics;
 
     /**
      * The coverage report instance representing the supplied coverage report
      * file.
      *
      * @var Report
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php:91]:
```suggestion
```

File: src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php [hunk 6]
```diff
@@ -81,20 +82,20 @@ class CrapIndexAnalyzer extends AbstractAnalyzer implements AggregateAnalyzer, A
      *
      * @var array<string, array<string, float>>
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/CrapIndexAnalyzer.php:83]:
```suggestion
     * @var array<string, array<string, float>>|null
```

File: src/main/php/PDepend/Metrics/Analyzer/DependencyAnalyzer.php [hunk 7]
```diff
@@ -75,6 +75,7 @@ class DependencyAnalyzer extends AbstractAnalyzer
      * @var array<string, ASTNamespace>
      */
     protected $nodeSet = [];
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/DependencyAnalyzer.php:77]:
```suggestion
    protected array $nodeSet = [];
```

File: src/main/php/PDepend/Metrics/Analyzer/DependencyAnalyzer.php [hunk 8]
```diff
@@ -95,7 +96,7 @@ class DependencyAnalyzer extends AbstractAnalyzer
      *
      * @var array<string, array<string, mixed>>
      */
-    private $nodeMetrics = null;
+    private $nodeMetrics;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/DependencyAnalyzer.php:99]:
```suggestion
    private ?array $nodeMetrics = null;
```

File: src/main/php/PDepend/Metrics/Analyzer/DependencyAnalyzer.php [hunk 9]
```diff
@@ -95,7 +96,7 @@ class DependencyAnalyzer extends AbstractAnalyzer
      *
      * @var array<string, array<string, mixed>>
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/DependencyAnalyzer.php:97]:
```suggestion
     * @var array<string, array<string, mixed>>|null
```

File: src/main/php/PDepend/Metrics/Analyzer/HierarchyAnalyzer.php [hunk 10]
```diff
@@ -149,7 +149,7 @@ class HierarchyAnalyzer extends AbstractAnalyzer implements AnalyzerFilterAware,
      *
      * @var array<string, array<string, int>>
      */
-    private $nodeMetrics = null;
+    private $nodeMetrics;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/HierarchyAnalyzer.php:152]:
```suggestion
    private ?array $nodeMetrics = null;
```

File: src/main/php/PDepend/Metrics/Analyzer/HierarchyAnalyzer.php [hunk 11]
```diff
@@ -149,7 +149,7 @@ class HierarchyAnalyzer extends AbstractAnalyzer implements AnalyzerFilterAware,
      *
      * @var array<string, array<string, int>>
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/HierarchyAnalyzer.php:150]:
```suggestion
     * @var array<string, array<string, int>>|null
```

File: src/main/php/PDepend/Metrics/Analyzer/InheritanceAnalyzer.php [hunk 12]
```diff
@@ -130,7 +130,7 @@ class InheritanceAnalyzer extends AbstractAnalyzer implements
      *
      * @var array<string, array<string, mixed>>
      */
-    private $nodeMetrics = null;
+    private $nodeMetrics;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/InheritanceAnalyzer.php:133]:
```suggestion
    private ?array $nodeMetrics = null;
```

File: src/main/php/PDepend/Metrics/Analyzer/InheritanceAnalyzer.php [hunk 13]
```diff
@@ -130,7 +130,7 @@ class InheritanceAnalyzer extends AbstractAnalyzer implements
      *
      * @var array<string, array<string, mixed>>
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/InheritanceAnalyzer.php:131]:
```suggestion
     * @var array<string, array<string, mixed>>|null
```

File: src/main/php/PDepend/Metrics/Analyzer/NodeCountAnalyzer.php [hunk 14]
```diff
@@ -112,7 +112,7 @@ class NodeCountAnalyzer extends AbstractAnalyzer implements AnalyzerFilterAware,
      *
      * @var array<string, array<string, int>>
      */
-    private $nodeMetrics = null;
+    private $nodeMetrics;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/NodeCountAnalyzer.php:115]:
```suggestion
    private ?array $nodeMetrics = null;
```

File: src/main/php/PDepend/Metrics/Analyzer/NodeCountAnalyzer.php [hunk 15]
```diff
@@ -112,7 +112,7 @@ class NodeCountAnalyzer extends AbstractAnalyzer implements AnalyzerFilterAware,
      *
      * @var array<string, array<string, int>>
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Metrics/Analyzer/NodeCountAnalyzer.php:113]:
```suggestion
     * @var array<string, array<string, int>>|null
```

File: src/main/php/PDepend/Report/Dependencies/Xml.php [hunk 16]
```diff
@@ -75,20 +75,21 @@ class Xml extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareGen
      *
      * @var ASTArtifactList<ASTNamespace>
      */
-    protected $code = null;
+    protected $code;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Dependencies/Xml.php:78]:
```suggestion
    protected ?ASTArtifactList $code = null;
```

File: src/main/php/PDepend/Report/Dependencies/Xml.php [hunk 17]
```diff
@@ -75,20 +75,21 @@ class Xml extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareGen
      *
      * @var ASTArtifactList<ASTNamespace>
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Dependencies/Xml.php:76]:
```suggestion
     * @var ASTArtifactList<ASTNamespace>|null
```

File: src/main/php/PDepend/Report/Dependencies/Xml.php [hunk 18]
```diff
@@ -75,20 +75,21 @@ class Xml extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareGen
...
      */
     protected $fileSet = [];
+
     /**
      * The log output file.
      *
      * @var string
      */
-    private $logFile = null;
+    private $logFile;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Dependencies/Xml.php:92]:
```suggestion
    private ?string $logFile = null;
```

File: src/main/php/PDepend/Report/Dependencies/Xml.php [hunk 19]
```diff
@@ -75,20 +75,21 @@ class Xml extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareGen
...
      * Set of all analyzed files.
      *
      * @var ASTCompilationUnit[]
      */
     protected $fileSet = [];
+
     /**
      * The log output file.
      *
      * @var string
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Dependencies/Xml.php:90]:
```suggestion
```

File: src/main/php/PDepend/Report/Jdepend/Chart.php [hunk 20]
```diff
@@ -70,21 +70,21 @@ class Chart extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareG
      *
      * @var string
      */
-    private $logFile = null;
+    private $logFile;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Jdepend/Chart.php:73]:
```suggestion
    private ?string $logFile = null;
```

File: src/main/php/PDepend/Report/Jdepend/Chart.php [hunk 21]
```diff
@@ -70,21 +70,21 @@ class Chart extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareG
      *
      * @var string
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Jdepend/Chart.php:71]:
```suggestion
```

File: src/main/php/PDepend/Report/Jdepend/Chart.php [hunk 22]
```diff
@@ -70,21 +70,21 @@ class Chart extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareG
...
-    private $logFile = null;
+    private $logFile;
 
     /**
      * The context source code.
      *
      * @var ASTArtifactList<ASTNamespace>
      */
-    private $code = null;
+    private $code;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Jdepend/Chart.php:80]:
```suggestion
    private ?ASTArtifactList $code = null;
```

File: src/main/php/PDepend/Report/Jdepend/Chart.php [hunk 23]
```diff
@@ -70,21 +70,21 @@ class Chart extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareG
      *
      * @var string
      */
-    private $logFile = null;
+    private $logFile;
 
     /**
      * The context source code.
      *
      * @var ASTArtifactList<ASTNamespace>
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Jdepend/Chart.php:78]:
```suggestion
     * @var ASTArtifactList<ASTNamespace>|null
```

File: src/main/php/PDepend/Report/Jdepend/Chart.php [hunk 24]
```diff
@@ -70,21 +70,21 @@ class Chart extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareG
...
-    private $code = null;
+    private $code;
 
     /**
      * The context analyzer instance.
      *
      * @var DependencyAnalyzer
      */
-    private $analyzer = null;
+    private $analyzer;
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Jdepend/Chart.php:87]:
```suggestion
    private ?DependencyAnalyzer $analyzer = null;
```

File: src/main/php/PDepend/Report/Jdepend/Chart.php [hunk 25]
```diff
@@ -70,21 +70,21 @@ class Chart extends AbstractASTVisitor implements CodeAwareGenerator, FileAwareG
...
      *
      * @var ASTArtifactList<ASTNamespace>
      */
-    private $code = null;
+    private $code;
 
     /**
      * The context analyzer instance.
      *
      * @var DependencyAnalyzer
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Report/Jdepend/Chart.php:85]:
```suggestion
```

File: src/main/php/PDepend/Metrics/Analyzer/CodeRankAnalyzer.php (hunk 1 above)
Comment (AJenbo) [reply src/main/php/PDepend/Metrics/Analyzer/CodeRankAnalyzer.php:116]: Hm I was thinking that most of these where set as a form of "invalid" or "Uninitialized" state.

To keep this PR simple, by just have rule based changes in it, I have removed the rule that causes this change. We can then revisit it individually at a later point.

Comment (AJenbo) [reply src/main/php/PDepend/Metrics/Analyzer/CodeRankAnalyzer.php]: Here we go :)

https://github.com/pdepend/pdepend/pull/806

**Comment by AJenbo** [comment]
Lol yeah that was a lot of changes :)

**Pull request merged**